code/object/waypoint.cpp

```diff
@@ -178,14 +184,13 @@ waypoint_list *find_matching_waypoint_list(const char *name)
 waypoint *find_matching_waypoint(const char *name)
 {
 	Assert(name != NULL);
-	SCP_list<waypoint_list>::iterator ii;
 
-	for (ii = Waypoint_lists.begin(); ii != Waypoint_lists.end(); ++ii)
+	for (auto &ii: Waypoint_lists)
```

Comment:
Can this loop be const, too?

Reply:
Can do

Reply:
Actually it can't, because the return pointer is not const (and the return pointer is used in non-const ways).